client/tests/integration/smartcontracts/validator_with_admin/src/lib.rs

```diff
@@ -116,34 +122,39 @@ impl ExpressionEvaluator for CustomValidator {
 }
 
 #[entrypoint]
-pub fn migrate() -> MigrationResult {
-    CustomValidator::migrate()
+pub fn migrate(block_height: u64) -> MigrationResult {
+    CustomValidator::migrate(block_height)
 }
 
 #[entrypoint]
 pub fn validate_transaction(
     authority: AccountId,
     transaction: VersionedSignedTransaction,
+    block_height: u64,
 ) -> Result {
-    let mut validator = CustomValidator(DefaultValidator::new());
+    let mut validator = CustomValidator(DefaultValidator::new(block_height));
 
     validator.visit_transaction(&authority, &transaction);
 
     validator.0.verdict
 }
 
 #[entrypoint]
-pub fn validate_instruction(authority: AccountId, instruction: InstructionBox) -> Result {
-    let mut validator = CustomValidator(DefaultValidator::new());
+pub fn validate_instruction(
+    authority: AccountId,
+    instruction: InstructionBox,
+    block_height: u64,
+) -> Result {
+    let mut validator = CustomValidator(DefaultValidator::new(block_height));
 
     validator.visit_instruction(&authority, &instruction);
 
     validator.0.verdict
 }
 
 #[entrypoint]
-pub fn validate_query(authority: AccountId, query: QueryBox) -> Result {
-    let mut validator = CustomValidator(DefaultValidator::new());
+pub fn validate_query(authority: AccountId, query: QueryBox, block_height: u64) -> Result {
```

Comment:
Seems like `authority`, `query` and `block_height` (and whatever will appear in future) should be packed into a single `ValidateQueryEntrypointPayload` struct. Same applies to other entrypoints (i.e. `migrate()`, `validate_instruction()` etc).

Reply:
Agree with @0x009922. We can pack this fields into the struct inside `#[entrypoint]` macro for example.

Reply:
- Created #3791